### PR TITLE
Major updates

### PR DESCRIPTION
**Pull request by mclintprojects**
Hi!

Here's what's changed:
- `subscribeToChannel` renamed to `subscribe`
- `unsubscribeToChannel` renamed to `unsubscribe`
- Changed `performAction` from `performAction(channel, actionName, {channelParams, actionParams})` to `performAction(channel, {action, channelParams, actionParams})`
- Renamed a few functions to be much more clear
- Add tests for `channel_id.dart`. I was unable to add tests for `action_cable.dart`.
- Updated the README
- Update the library's version number to 1.0.0 because of the breaking changes I introduced.

### CODE REVIEW

File: lib/action_cable.dart [hunk 1]
```diff
@@ -1,30 +1,28 @@
 import 'dart:async';
 import 'dart:convert';
-
 import 'package:web_socket_channel/io.dart';
-
 import 'channel_id.dart';
 
 typedef _OnConnectedFunction = void Function();
 typedef _OnChannelSubscribedFunction = void Function();
+typedef _OnChannelDisconnectedFunction = void Function();
```
Comment (bqmackay) [inline lib/action_cable.dart]: 👍

**Comment by Dreamersoul** [comment]
hello, this is getting merged? in my use case i need to show if the cable disconnects, thanks!

**Comment by namiwang** [comment]
sorry, just noticed this., will review and merge.

**Comment by namiwang** [comment]
@mclintprojects Hi! Thanks for your help!

Could you please ignore files/folders like `.dart_tool` and `.packages`.

ref: https://dart.dev/guides/libraries/private-files

**Comment by namiwang** [comment]
@mclintprojects 

I see a lot of formatting/linting commits.

For quotes, personally I'm using a single for `'static string'` and a double for `"${dynamic} string"`, but we could just use doubles all along.

I'm not that familiar with dart toolchains, are you using some linter? We could just include a linter config I guess.

**Comment by mclintprojects** [comment]
Thanks for getting to this. 

I'll ignore them.

Also, I use VSCode and I think it uses dartfmt under the hood because I personally didn't setup any linting/formatting toolchain.

**Comment by namiwang** [comment]
Hi I just updated `.gitignore` as suggested by https://dart.dev/guides/libraries/private-files, you could just merge that from master.

Thanks!

**Comment by mclintprojects** [comment]
Hi @namiwang, I just added a commit that removes the files you requested.

**Comment by namiwang** [comment]
@all-contributors please add @mclintprojects  for code and test

**Pull request merged**